frame/elections-phragmen/src/lib.rs

```diff
@@ -129,79 +129,118 @@ type NegativeImbalanceOf<T> =
 /// Migrations to version [`3.0.0`], as denoted by the changelog.
 pub mod migrations_3_0_0 {
 	use super::*;
+	use codec::FullCodec;
 	use frame_support::{
-		migration::StorageKeyIterator,
-		traits::{GetPalletVersion, PalletVersion},
+		storage::types::{StorageMap, StorageValue},
+		traits::{GetPalletVersion, PalletVersion, StorageInstance},
 		Twox64Concat,
 	};
 
-	pub fn apply<T: Config>(
-		old_voter_bond: BalanceOf<T>,
-		old_candidacy_bond: BalanceOf<T>,
+	type NewCandidates<I, A, B> = StorageValue<I, Vec<(A, B)>>;
+	type NewMembers<I, A, B> = StorageValue<I, Vec<SeatHolder<A, B>>>;
+	type NewRunnersUp<I, A, B> = StorageValue<I, Vec<SeatHolder<A, B>>>;
+	// NOTE: maybe be generic on the damn hasher as well, neh?
+	type NewVoting<I, A, B> = StorageMap<I, Twox64Concat, A, Voter<A, B>>;
+
+	/// Apply all of the migrations from 2_0_0 to 3_0_0.
+	///
+	/// ### Warning
+	///
+	/// This code will **ONLY** check that the storage version is less than 2_0_0. Further check
+	/// might be needed at the user runtime.
+	///
+	/// Be aware that this migration is intended to be used only for the mentioned versions. Use
+	/// with care and run at your own risk.
+	pub fn apply<
+		Instance: StorageInstance,
```

Comment:
`StorageInstance` holds the information of one storage. I.e. module prefix and storage prefix.
But here you use it for multiple storage.

For now both module_prefix and storage_prefix are fixed inside elections-phragmen so we shouldn't need to give any of these information.
Or we can give the argument `module_prefix`.